### PR TITLE
[clang][Modules] Fixing Build Breaks When -DLLVM_ENABLE_MODULES=ON

### PR DESCRIPTION
**Pull request by qiongsiwu**
A few recent changes are causing build breaks when `-DLLVM_ENABLE_MODULES=ON` (such as 834dfd23155351c9885eddf7b9664f7697326946 and 7dfdca1961aadc75ca397818bfb9bd32f1879248). 

This PR makes the required updates so that clang/llvm builds when `-DLLVM_ENABLE_MODULES=ON`.

### CODE REVIEW

File: clang/include/clang/Support/Compiler.h [hunk 1]
```diff
@@ -14,8 +14,6 @@
 #ifndef CLANG_SUPPORT_COMPILER_H
 #define CLANG_SUPPORT_COMPILER_H
 
-#include "llvm/Support/Compiler.h"
```
Comment (Bigcheese) [inline clang/include/clang/Support/Compiler.h:17]: The include of this file needs to move to Attr.h, module imports can't be in a namespace, so putting the include in Attrs.inc is wrong.

You should also add a new top level module for the Support directory.

`module Clang_Support { requires cplusplus umbrella "clang/Support" module * { export * } }`

File: llvm/tools/llvm-jitlink/llvm-jitlink-executor/CMakeLists.txt [hunk 2]
```diff
@@ -2,6 +2,7 @@ set(LLVM_LINK_COMPONENTS
   OrcShared
   OrcTargetProcess
   Support
+  Core
```
Comment (cyndyishida) [inline llvm/tools/llvm-jitlink/llvm-jitlink-executor/CMakeLists.txt:5]: Why did you need to add link dependencies?

Comment (qiongsiwu) [reply llvm/tools/llvm-jitlink/llvm-jitlink-executor/CMakeLists.txt:5]: Good question! @Bigcheese and myself discussed. We technically may not need `Core` here and I will revise. The dependency comes from https://github.com/llvm/llvm-project/blob/00e1cc4c9d002c78cf890b630343b052ebca0399/llvm/include/llvm/SandboxIR/Type.h#L286. Specifically, the new `SandBoxIR` code uses a function (`Type::print`) from `LLVMCore`.  Linking fails without the dependency.

The `SandBoxIR` code currently does not have module maps at all and I will go ahead and add them. Maybe we will no longer need the dependencies after the module maps are added.

File: clang/include/clang/Support/Compiler.h (hunk 1 above)
Comment (qiongsiwu) [reply clang/include/clang/Support/Compiler.h]: Thanks! Fixed!

File: llvm/tools/llvm-jitlink/llvm-jitlink-executor/CMakeLists.txt (hunk 2 above)
Comment (qiongsiwu) [reply llvm/tools/llvm-jitlink/llvm-jitlink-executor/CMakeLists.txt]: Code is revised to avoid adding the Core dependency.

**Review by cyndyishida — APPROVED** [review]
LGTM
